### DISCUSSION TITLE
TQG in S^1 \times [0,1]

### DISCUSSION THREAD

**Discussion by fberonvera** [CliMA/Oceananigans.jl#3478] [Computational science!]
Following Greg Wagner suggestion, I'm asking the community if there would be interest to collaborate with myself and my group in Miami in the implementation in S^1 x [0,1], a periodic zonal channel of the \beta-plane, of the system

q_t + [p, q - b] = 0
b_t + [p, b] = 0

where [,] is the canonical bracket in R^2 and

\Delta p - p = q - b - \beta y, 

subjected to the no-flow through the solid walls boundary condition:

p_x(x,0,t) = 0 = p_x(x,1,t)?

This system is known as the thermal QG model.  We can handle integration in T^2, but the walls require further care.  Eventually a more general domain would be nice to see this system implemented in as well as extensions I proposed to include stratification, always preserving the simple 2-d nature of the equations (and underlying Lie-Poisson geometry).

Thank you in advance.

Francisco Beron-Vera

**Comment by glwagner** [comment]
I believe there is a decent-sized literature that discusses how to implement the QG equation in a finite-volume framework. Here's a recent one that looks promising, since it could leverage our WENO advection schemes for tracers for PV advection:

https://archimer.ifremer.fr/doc/00817/92894/99308.pdf

The challenge of this scheme is implementing the Poisson solver, which differs from the currently-implemented suite of Poisson pressure solvers in two ways: 1) the C-grid location of the streamfunction (Face, Face) --- where (X, Y) denotes the staggered location of variables in x and y --- differs from pressure (Center, Center) and as a result the FFT transforms and boundary conditions are different, and 2) since PV is located at (Center, Center), there are some additional considerations associated with the reconstruction of the RHS at (Face, Face). That said, their method is quite similar to the ones we have so it does seems feasible.

**Comment by francispoulin** [comment]
Hello Francisco (@fberonvera), I have had much experience with the QG model and would be very happy to talk about what is needed to develop a QG model.  I have helped to develop the shallow water model and even though it is still experimental, I am currently working on making it more stable.

For testing, I might suggest we start with the barotropic QG model since it is only 2D that is so much faster to solve and we don't need to be too clever with the Poisson solver.  If you wanted to chat you can email me at fpoulin@uwaterloo.ca .  

Francis Poulin
